docker/ffmpeg_local/convert.py

```diff
@@ -0,0 +1,117 @@
+#! /usr/bin/env python
+
+import glob
+import json
+import os.path
+import subprocess
+import sys
+
+from re import compile
+
+GIRDER_WORKER_DIR = os.path.join('/', 'mnt', 'girder_worker', 'data')
+
+RE_VIDEO_INFO = compile(r'''^\s+Stream #.+?: Video''')
+RE_AUDIO_INFO = compile(r'''^\s+Stream #.+?: Audio''')
+RE_DURATION_INFO = compile(r'''^\s+Duration: ''')
+
+FFMPEG = 'ffmpeg'
+
+
+def duration_parse(durstr):
+    """Parse a duration of the form [[hh:]mm:]ss[.sss] and return a float
+     with the duration or None for failure to parse.
+    Enter: durstr: string with the duration.
+    Exit:  duration: duration in seconds."""
+    try:
+        durstr = durstr.strip().split(":")
+        dur = 0
+        for part in range(len(durstr)):
+            dur += float(durstr[-1-part])*(60**part)
+    except Exception:
+        return None
+    return dur
+
+
+subprocess.call(['ls', GIRDER_WORKER_DIR])
+sys.stdout.flush()
+
+input_file = next(glob.iglob(os.path.join(GIRDER_WORKER_DIR, 'input.*')))
+
+cmd = [FFMPEG, '-i', input_file]
```

Comment:
I had also considered bindings for ffmpeg, but forewent them mainly because I don't think they buy us anything technically.  It's probably also worth mentioning that according to the legal understanding of several folks (admittedly, none who are lawyers), using the binding might expose us to additional claims stemming from the precise definition of "linking" and/or "derivative works".  I, for one, disagree that there is any such distinction, but at least going command-line only gives these folks one less thing to worry about.